src/components/Transcript/Transcript.js

```diff
@@ -89,32 +285,21 @@ const Transcript = ({ playerID, manifestUrl, transcripts = [] }) => {
           "' on page. Transcript synchronization is disabled."
         );
       } else {
-        setPlayer(domPlayer.children[0]);
+        if (domPlayer.children[0]) playerRef.current = domPlayer.children[0];
+        else playerRef.current = domPlayer;
       }
-      if (player.current) {
-        let cIndex = parseInt(player.current.dataset['canvasindex']);
-        if (cIndex != canvasIndexRef.current) {
+
+      if (playerRef.current) {
+        let cIndex = parseInt(playerRef.current.dataset['canvasindex']);
+        if (Number.isNaN(cIndex)) cIndex = 0;
+        if (cIndex !== canvasIndexRef.current) {
           // Clear the transcript text in the component
           setTranscript([]);
-          setCanvasIndex(player.current.dataset['canvasindex']);
-
-          player.current.addEventListener('timeupdate', function (e) {
-            if (e == null || e.target == null) {
-              return;
-            }
-            const currentTime = e.target.currentTime;
-            textRefs.current.map((tr) => {
-              if (tr) {
-                const start = parseFloat(tr.getAttribute('starttime'));
-                const end = parseFloat(tr.getAttribute('endtime'));
-                if (currentTime >= start && currentTime <= end) {
-                  autoScrollAndHighlight(currentTime, start, end, tr);
-                } else {
-                  // remove highlight
-                  tr.classList.remove('active');
-                }
-              }
-            });
+          setCanvasIndex(cIndex);
+          setCurrentTime(playerRef.current.currentTime);
+
+          playerRef.current.addEventListener('timeupdate', () => {
+            setCurrentTime(playerRef.current.currentTime);
```

Comment:
This is not a blocker for this PR. This is more of a question for any future work, if we decide to not to make the transcript component an independent component.

Is it possible to remove this interval, which is used to keep track of the player instance on the page since the transcript component is not wrapped within the player context provider? 
The reason for this question is; with the work done in this PR the PlayerContext seems to be wrapping (at lease partially?) the transcript component, so we can use the player context to get this information instead of this interval? Am I understanding this correctly?

Reply:
IIRC, the `currentTime` property of `playerContext` is only updated when you use structured navigation (and possibly some other places) and does not update during playback.  You could certainly update that value during playback to make it the "source of truth" so that it is in sync with the player by adding something like this presumably to `VideoJSPlayer.js`:

```jsx
useEffect(() => {
  if (!playerRef.current) return;
  playerRef.current.addEventListener('timeupdate', () => {
    playerDispatch({ type: 'setCurrentTime' , currentTime: playerRef.current.currentTime })
  });
}, [playerRef.current]);
```

However that would have the undesirable effect of triggering a full re-render your entire tree several times a second even when the user is doing nothing. Since you're using react reducers and not redux you can't rely on selector memoization either... I'm not sure if hoisting the currentTime to the root is the best approach but even if you did, honestly, I don't think the performance would be too bad if it was denounced enough. A resolution of a quarter second (250ms debounce) should should be aggressive enough but I haven't done any testing to say for sure.

Reply:
IIRC, the `currentTime` property of `playerContext` is only updated when you use structured navigation (and possibly some other places) and does not update during playback.  You could certainly update that value during playback to make it the "source of truth" so that it is in sync with the player by adding something like this presumably to `VideoJSPlayer.js`:

```jsx
useEffect(() => {
  if (!playerRef.current) return;
  playerRef.current.addEventListener('timeupdate', () => {
    playerDispatch({ type: 'setCurrentTime' , currentTime: playerRef.current.currentTime })
  });
}, [playerRef.current]);
```

However that would have the undesirable effect of triggering a full re-render your entire tree several times a second even when the user is doing nothing. Since you're using react reducers and not redux you can't rely on selector memoization either... I'm not sure if hoisting the currentTime to the root is the best approach but even if you did, honestly, I don't think the performance would be too bad if it was denounced enough. A resolution of a quarter second (250ms debounce) should should be aggressive enough but I haven't done any testing to say for sure.

Reply:
Yes, that makes sense. Thanks for clarifying it. Apologies for the confusion.

In that case, I think the last change you made is not needed for this work. We can keep the Transcript component wrapped within the `PlayerContext` for this PR. 
And if we need to keep the Transcript component independent of the central state (need to discuss with the team), we can make the change as needed in the future.

So once you undo the last commit, I think we are ready to merge this PR 🎉

Reply:
If I revert that commit, the styleguide page will be broken because it isn't wrapped in context. The way I did the change in c7f9c08, it will check if it has dispatch via context, and if not, then it won't move the playhead when a line is clicked/won't markers to the progress bar but will otherwise work. So I can revert and make it have a hard fail if it's not wrapped but I don't see the benefit of that. 

that being said, I'll revert that commit momentarily

Reply:
Thanks!

Yes, makes sense to just do that at this stage. We could make changes as necessary in the future.

I checked the styleguide docs and they seem to be working with the latest code 👍